easybuild/tools/github.py

```diff
@@ -835,6 +877,136 @@ def is_patch_for(patch_name, ec):
     return soft_name
 
 
+def check_pr_eligible_to_merge(pr_data):
+    """
+    Check whether PR is eligible for merging.
+
+    :param pr_data: PR data obtained through GitHub API
+    :return: boolean value indicates whether PR is eligible
+    """
+    res = True
+
+    def not_eligible(msg):
+        """Helper function to warn about PR not being eligible for merging"""
+        print_msg("%s => not eligible for merging!" % msg, stderr=True, prefix=False)
+        return False
+
+    target = '%s/%s' % (pr_data['base']['repo']['owner']['login'], pr_data['base']['repo']['name'])
+    print_msg("Checking eligibility of %s PR #%s for merging..." % (target, pr_data['number']), prefix=False)
+
+    # check target branch, must be 'develop'
+    msg_tmpl = "* targets develop branch: %s"
+    if pr_data['base']['ref'] == 'develop':
+        print_msg(msg_tmpl % 'OK', prefix=False)
+    else:
+        res = not_eligible(msg_tmpl % "FAILED; found '%s'" % pr_data['base']['ref'])
+
+    # check test suite result, Travis must give green light
+    msg_tmpl = "* test suite passes: %s"
+    if pr_data['status_last_commit'] == 'success':
+        print_msg(msg_tmpl % 'OK', prefix=False)
+    elif pr_data['status_last_commit'] == 'pending':
+        res = not_eligible(msg_tmpl % "pending...")
+    elif pr_data['status_last_commit'] in ['error', 'failure']:
+        res = not_eligible(msg_tmpl % "FAILED")
+    else:
+        res = not_eligible(msg_tmpl % "(result unknown)")
+
+    # check for successful test report (checked in reverse order)
+    msg_tmpl = "* last test report is successful: %s"
+    test_report_regex = re.compile(r"^Test report by @\S+")
+    test_report_found = False
+    for comment in pr_data['issue_comments']['bodies'][::-1]:
+        if test_report_regex.search(comment):
+            if 'SUCCESS' in comment:
+                print_msg(msg_tmpl % 'OK', prefix=False)
+                test_report_found = True
+                break
+            elif 'FAILED' in comment:
+                res = not_eligible(msg_tmpl % 'FAILED')
+                test_report_found = True
+            else:
+                print_warning("Failed to determine outcome of test report for comment:\n%s" % comment)
+
+        if 'lgtm' in comment:
+            found_lgtm = True
+
+    if not test_report_found:
+        res = not_eligible(msg_tmpl % "(no test reports found)")
+
+    # check for style review by a human
+    found_lgtm = False
+    for comment in pr_data['issue_comments']['bodies']:
+        if 'lgtm' in comment:
+            found_lgtm = True
```

Comment:
I want to relax this a bit and also make this check pass when an approved review is submitted, especially since that's required now to merge (via the web interface).